src/Castle.Core/Castle.Core.csproj

```diff
@@ -3,7 +3,15 @@
 	<Import Project="..\..\buildscripts\common.props"></Import>
 
 	<PropertyGroup>
-		<TargetFrameworks>net462;netstandard2.0;netstandard2.1</TargetFrameworks>
+		<TargetFrameworks>net462;netstandard2.0;netstandard2.1;net6.0</TargetFrameworks>
+
+		<NoWarn>$(NoWarn);SYSLIB0003</NoWarn>
+		<!--
+			Code Access Security is not available in Core, and obsolete in net5.0+
+			This stops warnings about SYSLIB0003 (https://docs.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0003)
+			Depending on how these are deprecated in a future dotnet release, may need to wrap relevant code with "#if NET462_OR_GREATER" or similar instead.
+		-->
```

Comment:
(As above, I'm surprised this is necessary, and that we still have CAS code sticking around.)

Reply:
Same as above, change to pragma.